src/weights.jl

```diff
@@ -1,9 +1,9 @@
 
 ###### Weight vector #####
 
-immutable WeightVec{W,Vec<:RealVector}
-    values::Vec
-    sum::W
+immutable WeightVec{T, V<:RealVector, S} <: RealVector{T}
```

Comment:
I wondered about this too, then I realized that it *might* be useful to allow storing weights using a small type (`UInt8`, `Float16`) to save space, and yet be able to store the sum in a type that won't overflow or lose precision too easily (`Int64`, `BigInt`, `Float64`).